ppdet/modeling/backbones/dla.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import OrderedDict
+import paddle.fluid.layers as L
+
+from ppdet.core.workspace import register
+from ppdet.modeling.ops import Conv2dUnit
+
+
+def get_norm(norm_type):
+    assert norm_type in ['bn', 'sync_bn', 'gn', 'affine_channel']
+    bn = 0
+    gn = 0
+    af = 0
+    if norm_type == 'bn':
+        bn = 1
+    elif norm_type == 'sync_bn':
+        bn = 1
+    elif norm_type == 'gn':
+        gn = 1
+    elif norm_type == 'affine_channel':
+        af = 1
+    return bn, gn, af
+
+
+class BasicBlock(object):
+    def __init__(self, norm_type, inplanes, planes, stride=1, name=''):
+        super(BasicBlock, self).__init__()
+        bn, gn, af = get_norm(norm_type)
+        self.conv1 = Conv2dUnit(inplanes, planes, 3, stride=stride, bias_attr=False, bn=bn, gn=gn, af=af, act='relu', name=name+'.conv1')
+        self.conv2 = Conv2dUnit(planes, planes, 3, stride=1, bias_attr=False, bn=bn, gn=gn, af=af, act=None, name=name+'.conv2')
+        self.stride = stride
+
+    def __call__(self, x, residual=None):
+        if residual is None:
+            residual = x
+        out = self.conv1(x)
+        out = self.conv2(out)
+        out = L.elementwise_add(x=out, y=residual, act=None)
+        out = L.relu(out)
+        return out
+
+
+
+class Root(object):
+    def __init__(self, norm_type, in_channels, out_channels, kernel_size, residual, name=''):
+        super(Root, self).__init__()
+        bn, gn, af = get_norm(norm_type)
+        self.conv = Conv2dUnit(in_channels, out_channels, kernel_size, stride=1, bias_attr=False, bn=bn, gn=gn, af=af, act=None, name=name+'.conv')
```

Comment:
try `pip install pre-commit` and run `pre-commit run -s HEAD^ -o HEAD` to format code, code not format by `pre-commit` will failed in CI check